includes/support/abandoned-cart-hooks.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+function abandonedPluginHook($razorpayData) {
+
+     $rzpAbandonedData                                = array();
+     $rzpAbandonedData['woocommerceOrderID']          = $razorpayData['receipt'];
+     $rzpAbandonedData['amount']                      = $razorpayData['amount'];
+     $rzpAbandonedData['currency']                    = $razorpayData['currency'];
+     $rzpAbandonedData['customer_details']            = array();
+     $rzpAbandonedData['customer_details']['contact'] = $razorpayData['customer_details']['contact'];
+     $rzpAbandonedData['customer_details']['email']   = $razorpayData['customer_details']['email'];
+
+    $shippingAddress                                  = array(
+        'name'     => $razorpayData['customer_details']['shipping_address']['name'],
```

Comment:
Define the variable with the shipping address like
`$magicShippingAddress = $razorpayData['customer_details']['shipping_address']` and use this here. Same for the billing address.